core/src/main/java/com/datastax/oss/driver/internal/core/config/typesafe/TypesafeDriverConfig.java

```diff
@@ -62,6 +64,7 @@ public TypesafeDriverConfig(Config config) {
           new TypesafeDriverExecutionProfile.Base(entry.getKey(), entry.getValue()));
     }
     this.profiles = builder.build();
+    this.defaultProfile = profiles.get(DriverExecutionProfile.DEFAULT_NAME);
```

Comment:
Making this stateful doesn't impact config reloading.  Makes sense given the refresh() op in TypesafeDriverConfig.reload() but it seemed worth validating in an explicit test:

```java
  @Test
  public void should_update_default_profile_on_reload() {
    TypesafeDriverConfig config = parse("int1 = 42\n profiles { profile1 { int1 = 43 } }");
    assertThat(config.getDefaultProfile().getInt(MockOptions.INT1)).isEqualTo(42);
    config.reload(ConfigFactory.parseString("int1 = 44\n profiles { profile1 { int1 = 45 } }"));
    assertThat(config.getDefaultProfile().getInt(MockOptions.INT1)).isEqualTo(44);
  }
```

Might be a good idea to add something like this regardless.